src/domain/checklist/sendPaymentNotificationChecklist.ts

```diff
@@ -62,24 +58,24 @@ export const uploadToS3Check = {
   ),
 };
 
-// Send notification endpoint
-//  expect a request that produces a 2xx as a response
-//      the x-api-key header should be filled
 export const createNotificationRequestCheck = {
   group,
   name: 'Expect a send notification request that matches all the criteria',
-  eval: flow(
-    RA.filterMap(isNewNotificationRecord),
-    RA.some(
+  eval: pipe(
+    R.Do,
+    R.apS('preloadRecordList', RA.filterMap(isPreLoadRecord)),
+    R.apS('uploadToS3RecordList', RA.filterMap(isUploadToS3Record)),
+    R.apS('newNotificationRecordList', RA.filterMap(isNewNotificationRecord)),
+    R.map(({ preloadRecordList, uploadToS3RecordList, newNotificationRecordList }) =>
       pipe(
-        existsApiKey,
-        P.and(hasRecipientTaxId),
-        P.and(hasRecipientDigitalDomicile),
-        P.and(hasPhysicalAddress),
-        P.and(hasRegisteredLetterAsPhysicalDocumentType),
-        P.and(hasRecipientPaymentCreditorTaxId),
-        P.and(hasRecipientPaymentNoticeCode),
-        P.and(hasSuccessfulResponse)
+        newNotificationRecordList,
+        RA.some(
+          pipe(
+            matchNewNotificationRecordCriteria,
```

Comment:
Just a suggestion, the choice is yours
```suggestion
            existsApiKey,
            P.and(hasRecipientTaxId),
            P.and(hasRecipientDigitalDomicile),
            P.and(hasPhysicalAddress),
            P.and(hasRegisteredLetterAsPhysicalDocumentType),
            P.and(hasRecipientPaymentCreditorTaxId),
            P.and(hasRecipientPaymentNoticeCode),
            P.and(hasSuccessfulResponse),
```

Reply:
Done!